internal/services/servicebus/servicebus_namespace_disaster_recovery_config_data_source.go

```diff
@@ -94,6 +76,29 @@ func dataSourceServiceBusNamespaceDisasterRecoveryConfig() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FivePointOh() {
+		resource.Schema["namespace_id"].Required = false
+		resource.Schema["namespace_id"].Optional = true
+		resource.Schema["namespace_name"] = &pluginsdk.Schema{
+			Type:          pluginsdk.TypeString,
+			Optional:      true,
+			ValidateFunc:  validate.NamespaceName,
+			RequiredWith:  []string{"resource_group_name"},
```

Comment:
I think we want to keep the `AtLeastOneOf` tag over `RequiredWith` to keep the same functionality because the way it's written at the moment, you don't have to specify any of the fields to get Terraform to run. I doubt anyone wouldn't pass things in but extra guard rails don't hurt.

We can also add that AtLeastOneOf block a few lines up above to `namespace_id`

```
resource.Schema["namespace_id"].AtLeastOneOf = []string{"namespace_id", "resource_group_name", "namespace_name"}
```

Reply:
Maybe we could talk about it next week, my understanding is that for this data block to "work", it needs a `namespace_id` XOR ( `namespace_name` AND `resource_group_name` ). Using `AtLeastOneOf()` seems to give the impression that it doesn't work like that. I'm not sure i'm following you about not having to specify any fields, I've tested it, it only will plan if the above condition is satisfied.

Reply:
Maybe I'm starting to understand. If you're using 5.0, it requires `namespace_id`, and no longer recognizes the other two, but if you're using < 5.0, it allows using nothing! Is that the idea?